CRM/Civigiftaid/Hook/Post/SetContributionGiftAidEligibility.php

```diff
@@ -0,0 +1,168 @@
+<?php
+
+use CRM_Civigiftaid_Utils_GiftAid as GiftAidUtil;
+
+/**
+ * Class CRM_Civigiftaid_Hook_Post_SetContributionGiftAidEligibility.
+ */
+class CRM_Civigiftaid_Hook_Post_SetContributionGiftAidEligibility {
+
+  /**
+   * Set the gift aid eligibility for a contribution.
+   *
+   * @param string $op
+   *   The operation being performed.
+   * @param string $objectName
+   *   Object name.
+   * @param mixed $objectId
+   *   Object ID.
+   * @param object $objectRef
+   *   Object reference.
+   */
+  public function run($op, $objectName, $objectId, &$objectRef) {
+    if (!$this->shouldRun($op, $objectName)) {
+      return;
+    }
+
+    $this->setGiftAidEligibilityStatus($objectId);
+  }
+
+  /**
+   * Sets gift aid eligibility status for a contribution.
+   *
+   * We are mainly concerned about contribution added from Events
+   * or the membership pages by the admin and not the Add new contribution screen as
+   * this screen already has a form widget to set gift aid eleigibility status.
+   *
+   * This function checks if the contribution is eligible and automatically sets
+   * the status to yes, else, it sets it to No.
+   *
+   * @param int $contributionId
+   *   Contribution Id.
+   */
+  private function setGiftAidEligibilityStatus($contributionId) {
+    $currentPath = CRM_Utils_System::currentPath();
+    $addMembershipPath = 'civicrm/member/add';
+    $registerEventParticipantPath = 'civicrm/participant/add';
+    if (!in_array($currentPath, [$registerEventParticipantPath, $addMembershipPath])) {
+      return;
+    }
+
+    $eligibilityFieldId = $this->getEligibilityFieldId();
+    $eligibilityField = 'custom_' . $eligibilityFieldId;
+
+    $contribution = civicrm_api3('Contribution', 'getsingle', [
+      'return' => ['financial_type_id', 'contact_id'],
+      'id' => $contributionId,
+    ]);
+
+    $allFinancialTypesEnabled = CRM_Civigiftaid_Form_Admin::isGloballyEnabled();
+    if ($allFinancialTypesEnabled || $this->financialTypeIsEligible($contribution['financial_type_id'])) {
+      $eligibility = GiftAidUtil::DECLARATION_IS_YES;
+    }
+    else {
+      $eligibility = GiftAidUtil::DECLARATION_IS_NO;
+    }
+
+    civicrm_api3('Contribution', 'create', [
+      $eligibilityField => $eligibility,
+      'id' => $contributionId,
+    ]);
+
+    $contributionContact = $contribution['contact_id'];
+    if (!GiftAidUtil::getDeclaration($contributionContact) && $eligibility == GiftAidUtil::DECLARATION_IS_YES) {
+      CRM_Core_Session::setStatus(ts($this->getMissingGiftAidDeclarationMessage($contributionContact)), 'Gift Aid Declaration', 'success');
+    }
+  }
+
+  /**
+   * Returns a warning message about missing gift aid declaration for
+   * contribution contact.
+   *
+   * @param int $contactId
+   *   Contact Id.
+   *
+   * @return string
+   *
+   */
+  private function getMissingGiftAidDeclarationMessage($contactId) {
+    $message = "This contribution has been automatically marked as Eligible for Gift Aid.
```

Comment:
@tunbola  Shall we add localisation on the message?

Reply:
Good point sir

Reply:
Actually it is done here: https://github.com/compucorp/uk.co.compucorp.civicrm.giftaid/blob/82266adf5b31f227d28d3fa74b2ff1ef3cced9e0/CRM/Civigiftaid/Hook/Post/SetContributionGiftAidEligibility.php#L74